tests/plugins/test_reactor_config.py

```diff
@@ -273,3 +274,58 @@ def test_good_cluster_config(self, tmpdir, config, clusters):
         enabled = conf.get_enabled_clusters_for_platform('platform')
         assert set([(x.name, x.max_concurrent_builds)
                     for x in enabled]) == set(clusters)
+
+    @pytest.mark.parametrize(('default', 'is_valid'), (
+        ('release', True),
+        ('beta', True),
+        ('unsigned', True),
+        ('spam', False),
+    ))
+    def test_odcs_config(self, tmpdir, default, is_valid):
+        filename = str(tmpdir.join('config.yaml'))
+        with open(filename, 'w') as fp:
+            fp.write(dedent("""\
+                version: 1
+                odcs:
+                   signing_intents:
+                   - name: release
+                     keys: [R123]
+                   - name: beta
+                     keys: [R123, B456]
+                   - name: unsigned
+                     keys: []
+                   default_signing_intent: {default}
+                """.format(default=default)))
+
+        tasker, workflow = self.prepare()
+        plugin = ReactorConfigPlugin(tasker, workflow, config_path=str(tmpdir))
+        assert plugin.run() is None
+
+        if not is_valid:
+            with pytest.raises(ValueError):
+                get_config(workflow).get_odcs_config()
+            return
+
+        odcs_config = get_config(workflow).get_odcs_config()
+
+        assert odcs_config.default_signing_intent == default
+
+        assert odcs_config.signing_intents == [
+            {'name': 'unsigned', 'keys': [], 'restrictiveness': 0},
+            {'name': 'beta', 'keys': ['R123', 'B456'], 'restrictiveness': 1},
+            {'name': 'release', 'keys': ['R123'], 'restrictiveness': 2},
+        ]
+
+        with pytest.raises(ValueError):
```

Comment:
Tests starting at this line don't seem to depend on the parametrized values. Should they move to a separate test method?

Reply:
Agreed. Moved "invalid signing intent" to its own test.